tests/features/collection/newsletter_subscription.feature

```diff
@@ -0,0 +1,124 @@
+    Then I should not see the newsletter subscription form in the last tile
+
+    # If "Enable newsletter subscriptions" is not checked then it should be
+    # possible to submit the form without entering data in the two newsletter
+    # fields.
+    When I click the contextual link "Edit" in the Header region
+    And I uncheck "Enable newsletter subscriptions"
+    And I press "Publish"
+    Then I should see the heading "Volkor X"
+
+    # If "Enable newsletter subscriptions" is checked then the fields become
+    # required and an error message should be shown if they are not filled in.
+    When I click the contextual link "Edit" in the Header region
+    And I check "Enable newsletter subscriptions"
+    And I press "Publish"
+    Then I should see the following error messages:
+      | error messages                                                                     |
+      | Universe acronym field is required when newsletter subscriptions are enabled.      |
+      | Newsletter service ID field is required when newsletter subscriptions are enabled. |
+
+    # Fill in the information to connect with the Newsroom newsletter service.
+    # This should make it possible to submit the form without errors.
+    When I fill in "Universe acronym" with "volkor-x"
+    And I fill in "Newsletter service ID" with "123"
+    And I press "Publish"
+    Then I should see the heading "Volkor X"
+
+    # Now the subscription form should show up.
+    And I should see the newsletter subscription form in the last tile
+
+    # Disable the newsletter subscriptions again. This should make the form
+    # disappear.
+    When I click the contextual link "Edit" in the Header region
+    And I uncheck "Enable newsletter subscriptions"
+    And I press "Publish"
+    Then I should see the heading "Volkor X"
+    But I should not see the newsletter subscription form in the last tile
+
+  @javascript @newsroom_newsletter
+  Scenario: Subscribe to a newsletter
+    Given I am logged in as "Magdalini Kokinos"
+    When I go to the "Volkor X" collection
+    And I click the contextual link "Edit" in the Header region
+    And I click the "Additional fields" tab
+    And I check "Enable newsletter subscriptions"
+    And I fill in "Universe acronym" with "volkor-x"
+    And I fill in "Newsletter service ID" with "123"
+    And I press "Publish"
+    Then I should see the heading "Volkor X"
+
+    # Now the subscription form should show up.
+    And I should see the newsletter subscription form in the last tile
+
+    # The email address of the logged in user should be prefilled.
+    And the "E-mail address" field should contain "m.kokinos@example.com"
+
+    # When the user subscribes a success message should be shown.
+    When I press "Subscribe"
+    And I wait for AJAX to finish
+    Then I should see the following success messages:
+      | success messages                             |
+      | Thank you for subscribing to our newsletter. |
```

Comment:
We can't set it to the page since then we have to refresh the entire page, and the request was to have AJAX. I think this can be styled in the front end. The positioning of the error message can be controlled in the frontend and there is no reason it should have such a huge margin around it.

Also, how often do you think people will actually see this message? I guess most people will be able to enter their e-mail address correctly, or just submit the prefilled one.

I also think that this form will not remain in a tile forever. This location is far from optimal, I think our users will complain that nobody subscribes since this form is kind of hidden among the search results.